hazelcast/src/main/java/com/hazelcast/client/impl/connection/tcp/TcpClientConnection.java

```diff
@@ -314,12 +303,24 @@ public String toString() {
                 + ", lastReadTime=" + timeToStringFriendly(lastReadTimeMillis())
                 + ", lastWriteTime=" + timeToStringFriendly(lastWriteTimeMillis())
                 + ", closedTime=" + timeToStringFriendly(closedTime.get())
-                + ", connected server version=" + connectedServerVersion
                 + '}';
     }
 
-    public void setConnectedServerVersion(String connectedServerVersion) {
-        this.connectedServerVersion = connectedServerVersion;
+    /**
+     * Returns a string that would be less descriptive than the one returned
+     * from the {@code toString} method, but still would contain enough
+     * information to identify the connection.
+     * <p>
+     * This method is mainly intended to be used for logging, where the
+     * {@code toString} method would clutter the logging text due to
+     * its length.
+     */
+    public String toLessDescriptiveString() {
```

Comment:
do we really need to introduce this method? How much text differ in log?

Reply:
To me, log messages become very long without this method. 
With it
```
INFO: hz.client_1 [dev] [5.3.0-SNAPSHOT] Successfully connected to Alto channel NioChannel{/192.168.1.61:57031->/192.168.1.61:11006} for ClientConnection{connectionId=1, remoteAddress=[192.168.1.61]:5701, alive=true}

INFO: hz.client_1 [dev] [5.3.0-SNAPSHOT] All Alto channel connections are established for the ClientConnection{connectionId=1, remoteAddress=[192.168.1.61]:5701, alive=true}
```

Without it
```
INFO: hz.client_1 [dev] [5.3.0-SNAPSHOT] Successfully connected to Alto channel NioChannel{/192.168.1.61:56999->/192.168.1.61:11009} for ClientConnection{alive=true, connectionId=1, channel=NioChannel{/127.0.0.1:56989->/127.0.0.1:5701}, remoteAddress=[192.168.1.61]:5701, lastReadTime=2023-02-21 19:58:20.936, lastWriteTime=2023-02-21 19:58:20.932, closedTime=never}

INFO: hz.client_1 [dev] [5.3.0-SNAPSHOT] All Alto channel connections are established for the ClientConnection{alive=true, connectionId=1, channel=NioChannel{/127.0.0.1:56989->/127.0.0.1:5701}, remoteAddress=[192.168.1.61]:5701, lastReadTime=2023-02-21 19:58:20.936, lastWriteTime=2023-02-21 19:58:20.948, closedTime=never}
```

If you really feel like we don't need this, I can remove it.

Reply:
I would rather prefer all the info since usually logs are used for debugging purposes.

Reply:
OK, removed it in https://github.com/hazelcast/hazelcast/pull/23302/commits/058715cb28f5d3f458684e8a3af65dc8b6e7e0c2